### PR TITLE
Update user stories and other minor fixes

### LINKED ISSUE #144: Missing user stories
Some features not mentioned include (taken from release notes)
I want to...
- search elderly by nric
- mark calls with a specific date
- view the next contact date for each elderly

### LINKED ISSUE #145: Standardise constant naming and ug/dg formatting
- User guide: PHONE vs PHONE_NUMBER
- Help window/User guide: CALL FREQUENCY vs CALL_FREQUENCY

Other minor formatting fixes:
- Dev guide: font for `Implement a feature to clear the command history.` is a bit too big
- Glossary inconsistent bolding

### LINKED ISSUE #146: Misleading error message for invalid index
Currently error message reads `The person index provided is invalid` when index >= list size. This could be misleading since `mark 0` is also technically an invalid index but doesn't print this message but rather prints `Invalid command format!...` instead (due to checking of non zero positive integer).

Change error message to `The person index provided is out of bounds from the end of the list` to represent error when index >= list size

### LINKED ISSUE #149: Standardise use of elderly in ug
Probably should standardise using `elderly` instead of `person` to make it more user-orientated in the user guide
(e.g. mark the elderly as called vs mark the person as called)

### LINKED ISSUE #150: Inconsistent text and diagram in dg
The mark sequence diagram date was updated but the text in dg remained as `execute("mark 1 d/2020-01-01")` instead of `execute("mark 1 d/2024-10-25")`

### PR DESCRIPTION
**Pull request by linette-g**
Refer to issues for the specific fixes (just some minor changes all round, main update is the user stories)

User stories added:
- As a user, I want to search elderly by Nric, so that I can find this specific person given their Nric                                                    
- As a forgetful user, I want to mark calls with a specific date so that I can mark a call that I previously forgot to mark                                - As a user, I want to view the next contact date for each elderly, so that I can quickly see which day the next call should be made to better plan my work

Closes #144
Closes #145 
Closes #146 
Closes #149 
Closes #150

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -334,42 +335,45 @@ The following activity diagram summarizes what happens when a user interacts wit
...
+ | `****` | New user | list elderly contacts by priority (prioritised by last called date) | I know whom to contact first          | 
  | `****` | Frequent User | Remove elderly from the call list | Any elderly who has passed away or left the program will no longer be on the list | 
- | `****` | User | Record details of the elderly (NRIC etc.) | I know who I'm calling | 
+ | `****` | User | Record details of the elderly (NRIC etc.) | I know who I'm calling                | 
  | `****` | Frequent User | Add new elderly who have joined the Befriending Program | I can keep track of these new elderly and call them regularly | 
  | `****` | Expert User | Take notes regarding the call | I can keep track of things which should be followed up on | 
  | `****` | Frequent User | Update elderly contact information | I can keep the elderly information up to date | 
  | `****` | Frequent User | Search elderly by name | I can find this specific person if they were to call me and I can log it as a call | 
+ | `****` | User | Search elderly by Nric | I can find this specific person given their Nric |
```
Comment (zyng233) [inline docs/DeveloperGuide.md:348]: Maybe can change the person here to elderly to be more specified?

File: src/main/java/seedu/address/logic/Messages.java [hunk 2]
```diff
@@ -14,7 +14,8 @@ public class Messages {
 
     public static final String MESSAGE_UNKNOWN_COMMAND = "Unknown command";
     public static final String MESSAGE_INVALID_COMMAND_FORMAT = "Invalid command format! \n%1$s";
-    public static final String MESSAGE_INVALID_PERSON_DISPLAYED_INDEX = "The person index provided is invalid";
+    public static final String MESSAGE_INVALID_PERSON_DISPLAYED_INDEX = "The person index provided is out of bounds "
+            + "from the end of the list";
```
Comment (zyng233) [inline src/main/java/seedu/address/logic/Messages.java:18]: Not sure whether do we hv to standardise the use of elderly in Message also...

**Review by zyng233 — APPROVED** [review]
LGTM, just the standardization of elderly may need further discussion.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -334,42 +335,45 @@ The following activity diagram summarizes what happens when a user interacts wit
...
  | `***` | Expert User | Set multiple contacts as emergency contacts | I can call their next of kin in the event the elderly do not pick up or needs help | 
  | `***` | Frequent User | Export a specific elderly's details and history (All information) | So I can share it with another employee | 
  | `*` | Expert user | Be notified by the app when too many elderly calls are scheduled on one day | I will not overwork myself or spend too little time calling each elderly | 
  | `*` | Expert User | I can see a dashboard of my weekly or monthly call stats | I can track my productivity and ensure that I hit my KPI | 
  | `*` | Expert User | Set up automated messages for elderly contacts that are not reachable | I can have an alternative communication method | 
- | `*` | Expert User | Use AI to calculate the priority list for elderly based on their information | The correct eldelry are being prioritised  | 
+ | `*` | Expert User | Use AI to calculate the priority list for elderly based on their information | The correct eldelry are being prioritised |
```
Comment (brendanng7) [inline docs/DeveloperGuide.md:376]: looks like there is a typo on elderly here

File: src/main/java/seedu/address/logic/Messages.java (hunk 2 above)
Comment (linette-g) [reply src/main/java/seedu/address/logic/Messages.java:18]: Sure I'll do that

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (linette-g) [reply docs/DeveloperGuide.md:376]: Thanks nice catch

**Pull request merged**